homeassistant/util/distance.py

```diff
@@ -34,19 +47,35 @@ def convert(value: float, unit_1: str, unit_2: str) -> float:
 
     if unit_1 == LENGTH_MILES:
```

Comment:
For a future PR, instead of doing hundreds of `if…elif` statements, we should rewrite this like this:

```python
TO_METERS = {
  LENGTH_METERS: lambda meters: meters,
  LENGTH_YARD: lambda yards: yards * 0.9144,
  LENGTH_FEET: lambda feet: feet * 0.3048,
  ...
}

meters = TO_METERS[unit_1](value)
```

Reply:
Done in https://github.com/home-assistant/core/pull/43107